allennlp/tests/semparse/executors/wikitables_variable_free_executor_test.py

```diff
@@ -0,0 +1,383 @@
+        logical_form = """(select (same_as (filter_in all_rows fb:row.row.league a_league)
+                                   fb:row.row.playoffs)
+                           fb:row.row.league)"""
+        cell_list = self.executor.execute(logical_form)
+        assert cell_list == ["fb:cell.usl_a_league", "fb:cell.usl_first_division"]
+
+    def test_execute_works_with_sum(self):
+        # Get total "avg attendance".
+        logical_form = """(sum all_rows fb:row.row.avg_attendance)"""
+        sum_value = self.executor.execute(logical_form)
+        assert sum_value == 13197
+        # Total "avg attendance" where "playoffs" has "quarterfinals"
+        logical_form = """(sum (filter_in all_rows fb:row.row.playoffs quarterfinals)
+                                fb:row.row.avg_attendance)"""
+        sum_value = self.executor.execute(logical_form)
+        assert sum_value == 13197
+
+    def test_execute_works_with_average(self):
+        # Get average "avg attendance".
+        logical_form = """(average all_rows fb:row.row.avg_attendance)"""
+        avg_value = self.executor.execute(logical_form)
+        assert avg_value == 6598.5
+        # Average "avg attendance" where "playoffs" has "quarterfinals"
+        logical_form = """(average (filter_in all_rows fb:row.row.playoffs quarterfinals)
+                                fb:row.row.avg_attendance)"""
+        avg_value = self.executor.execute(logical_form)
+        assert avg_value == 6598.5
+
+    def test_execute_works_with_diff(self):
+        # Difference in "avg attendance" between rows with "usl_a_league" and "usl_first_division"
+        # in "league" columns.
+        logical_form = """(diff (filter_in all_rows fb:row.row.league usl_a_league)
+                                (filter_in all_rows fb:row.row.league usl_first_division)
+                                fb:row.row.avg_attendance)"""
+        avg_value = self.executor.execute(logical_form)
+        assert avg_value == 1141
+
+    def test_execute_fails_with_diff_on_non_numerical_columns(self):
+        logical_form = """(diff (filter_in all_rows fb:row.row.league usl_a_league)
+                                (filter_in all_rows fb:row.row.league usl_first_division)
+                                fb:row.row.league)"""
+        with self.assertRaises(ExecutionError):
+            self.executor.execute(logical_form)
+
+    def test_execute_fails_with_non_int_dates(self):
+        logical_form = """(date 2015 1.5 1)"""
+        with self.assertRaises(ExecutionError):
+            self.executor.execute(logical_form)
+
+    def test_date_comparison_works(self):
+        assert Date(2013, 12, 31) > Date(2013, 12, 30)
+        assert Date(2013, 12, 31) == Date(2013, 12, -1)
+        assert Date(2013, -1, -1) >= Date(2013, 12, 31)
+        assert Date(2013, 12, 31) != 2013
+        # pylint: disable=singleton-comparison
+        assert (Date(2013, 12, -1) > Date(2013, 12, 31)) == False
```

Comment:
You can remove the pylint statement and these parentheses if you do `Date(...) > Date(...) is False`.

Reply:
Done, thanks!

Reply:
Hmm.. looks like `>` and `is` have the same level of precedence, and since they're evaluated left to right, that wouldn't work. Keeping the original code.